src/main/java/com/android/volley/toolbox/DiskBasedCacheUtility.java

```diff
@@ -236,6 +236,15 @@ static String readString(CountingInputStream cis) throws IOException {
         return new String(b, "UTF-8");
     }
 
+    static String readString(ByteBuffer buffer) throws IOException {
```

Comment:
I think it's a bit strange for this to throw IOException - I think the only thing it can throw is UnsupportedEncodingException in new String, but this wouldn't be thrown by anything else in this method, right?

I wouldn't clean this up in this PR since it's a pattern we're taking from elsewhere, but in a follow-up PR, I would suggest perhaps adding a new method newUtf8String which catches UnsupportedEncodingException and rethrows it as a RuntimeException. UTF-8 should always exist so we shouldn't have to clutter code with handling in case it doesn't. Then callers won't have to deal with IOExceptions here - just the BufferUnderflowException if we run out of buffer.

Reply:
Also, since all these methods require a higher API, we can replace "UTF-8" with StandardCharsets.UTF-8 and that doesn't throw any exception. It wouldn't be the same as code for DiskBasedCache, but might be better since there'd be no error handling needed.